javascript/ql/lib/semmle/javascript/frameworks/Spife.qll

```diff
@@ -0,0 +1,416 @@
+    }
+
+    override predicate definesHeaderValue(string headerName, DataFlow::Node headerValue) {
+      exists(string header |
+        this.getAHeaderSource().hasPropertyWrite(header, headerValue) and
+        headerName = header.toLowerCase()
+      )
+    }
+
+    override DataFlow::Node getNameNode() {
+      result = this.getAHeaderSource().getAPropertyWrite().getPropertyNameExpr().flow()
+    }
+
+    override RouteHandler getRouteHandler() { result = reply.getRouteHandler() }
+  }
+
+  /**
+   * A header produced by a route handler with no explicit declaration of a Content-Type.
+   */
+  private class ContentTypeRouteHandlerHeader extends Http::ImplicitHeaderDefinition,
+    DataFlow::FunctionNode instanceof RouteHandler {
+    override predicate defines(string headerName, string headerValue) {
+      headerName = "content-type" and headerValue = "application/json"
+    }
+
+    override Http::RouteHandler getRouteHandler() { result = this }
+  }
+
+  /**
+   * An HTTP cookie defined in a Spife HTTP response.
+   */
+  private class CookieDefinition extends Http::CookieDefinition, DataFlow::MethodCallNode instanceof ReplySource {
+    CookieDefinition() {
+      // reply.cookie(RESPONSE, 'TEST', 'FOO', {"maxAge": 1000, "httpOnly": true, "secure": true})
+      this.ref().(DataFlow::MethodCallNode).getMethodName() = "cookie"
+    }
+
+    override DataFlow::Node getNameArgument() { result = this.getArgument(1) }
+
+    override DataFlow::Node getValueArgument() { result = this.getArgument(2) }
+
+    override RouteHandler getRouteHandler() { result = this.getRouteHandler() }
+  }
+
+  /**
+   * A response argument passed to the `reply` method.
+   */
+  private class ReplyArgument extends Http::ResponseSendArgument, DataFlow::Node {
+    RouteHandler rh;
+
+    ReplyArgument() {
+      exists(ReplySource reply, DataFlow::CallNode call |
+        reply.ref() = call and
+        call.getCalleeName() =
+          ["reply", "cookie", "link", "header", "headers", "raw", "status", "toStream", "vary"] and
+        this = call.getArgument(0) and
+        rh = reply.getRouteHandler()
+      )
+      or
+      this = rh.getAReturn()
+    }
+
+    override RouteHandler getRouteHandler() { result = rh }
+  }
+
+  /**
+   * An expression passed to the `template` method of the reply object
+   * as the value of a template variable.
+   */
+  private class TemplateInput extends Http::ResponseBody {
+    TemplateObjectInput obj;
+
+    TemplateInput() {
+      obj.getALocalSource().(DataFlow::ObjectLiteralNode).hasPropertyWrite(_, this)
+    }
+
+    override RouteHandler getRouteHandler() { result = obj.getRouteHandler() }
+  }
+
+  /**
+   * An object passed to the `template` method of the reply object.
+   */
+  private class TemplateObjectInput extends DataFlow::Node {
+    ReplySource reply;
+
+    TemplateObjectInput() {
+      exists(DataFlow::MethodCallNode call |
+        reply.ref() = call and
+        call.getMethodName() = "template" and
```

Comment:
The same as above. Use `getAMethodCall()`.  

You are not testing this class, otherwise you would have found this.  
I haven't looked into why that happens.

Reply:
I think I can simplify this one by using the `TemplateCall` defined below and just get the arg 0